libs/redis/langchain_redis/cache.py

```diff
@@ -701,3 +716,200 @@ async def aupdate(
+    ``pip install 'langchain-redis[langcache]'`` or
+    ``pip install 'langcache>=0.10.0'``.
+
+    Args:
+        embeddings (Optional[Embeddings]): Ignored. LangCache computes embeddings
+            server-side.
+        distance_threshold (float): Maximum distance for semantic matches.
+        ttl (Optional[int]): Cache TTL in seconds. If None, entries do not expire.
+        name (Optional[str]): Human-friendly cache name. Defaults to "llmcache".
+        prefix (Optional[str]): Prefix for namespacing. Defaults to "llmcache".
+        server_url (Optional[str]): LangCache API endpoint.
+        api_key (Optional[str]): API key for LangCache authentication.
+        cache_id (Optional[str]): Required LangCache instance identifier.
+        use_exact_search (bool): Enable exact match search. Defaults to True.
+        use_semantic_search (bool): Enable semantic search. Defaults to True.
+        distance_scale (Literal["normalized","redis"]): Distance scaling mode.
+        **kwargs: Additional options. Any ``vectorizer`` is ignored with a warning.
+
+    Example:
+        .. code-block:: python
+
+            from langchain_redis import LangCacheSemanticCache
+
+            cache = LangCacheSemanticCache(
+                cache_id="your-cache-id",
+                api_key="your-api-key",
+                name="mycache",
+                prefix="prod",
+                ttl=3600,
+            )
+
+    Notes:
+        - LangCache uses ``name`` as the cache identifier. To support ``prefix`` for
+          multi-tenant isolation, we map name/prefix:
+          * If both are provided and different, we use "name:prefix".
+          * If only ``prefix`` is meaningfully set, we use ``prefix``.
+          * Otherwise we use ``name``.
+        - Client-side vectorizers are ignored. Embeddings are computed server-side.
+        - Per-entry TTL is ignored; cache-level TTL applies if set.
+    """
+
+    # Hint for type checkers/IDEs; defined at class scope
+    cache: TC_LangCacheSemanticCache
+
+    def __init__(
+        self,
+        embeddings: Optional[Embeddings] = None,
+        distance_threshold: float = 0.2,
+        ttl: Optional[int] = None,
+        name: Optional[str] = _DEFAULT_CACHE_NAME,
+        prefix: Optional[str] = _DEFAULT_CACHE_PREFIX,
+        *,
+        server_url: Optional[str] = None,
+        api_key: Optional[str] = None,
+        cache_id: Optional[str] = None,
+        use_exact_search: bool = True,
+        use_semantic_search: bool = True,
+        distance_scale: Literal["normalized", "redis"] = "normalized",
+        **kwargs: Any,
+    ):
+        if not cache_id:
+            raise ValueError("cache_id is required for LangCacheSemanticCache")
+        if not api_key:
+            raise ValueError("api_key is required for LangCacheSemanticCache")
+
+        # LangCache uses 'name' as the identifier for the cache instance.
+        # To support the 'prefix' parameter for multi-tenant isolation,
```

Comment:
FYI the prefix here for langcache does nothing. It is used for the required `name` param within the redisvl LLMCache base class. But not used by langcache at all.

Reply:
Ah, yeah, I was originally thinking it would be beneficial if this was swappable with a `RedisSemanticCache` without changes, but it's probably best to avoid pinning this cache to the Redis cache's interface, other than the base class.